RR.App/Controllers/HRIS/EmployeeQualificationController.cs

```diff
@@ -23,107 +24,46 @@ public EmployeeQualificationController(AuthorizeIdentity identity, IEmployeeQual
     [HttpPost]
     public async Task<IActionResult> SaveEmployeeQualification([FromBody] EmployeeQualificationDto employeeQualificationDto)
     {
-        try
-        {
-            if (_identity.Role is "SuperAdmin" or "Admin" or "Talent" or "Journey")
-            {
-                var newQualification = await _employeeQualificationService.SaveEmployeeQualification(employeeQualificationDto, employeeQualificationDto.EmployeeId);
-                return Ok(newQualification);
-            }
-            
-            if (employeeQualificationDto.EmployeeId == _identity.EmployeeId)
-            {
-                var newQualification = await _employeeQualificationService.SaveEmployeeQualification(employeeQualificationDto, employeeQualificationDto.EmployeeId);
-                return Ok(newQualification);
-            }
-            return NotFound("User data being accessed does not match user making the request.");
-        }
-        catch (Exception ex)
-        {
-            return StatusCode(500, ex.Message);
-        }
+        if (_identity.Role is not ("SuperAdmin" or "Admin" or "Talent" or "Journey") && employeeQualificationDto.EmployeeId != _identity.EmployeeId)
+            throw new CustomException("Unauthorized Access.");
+
+        var newQualification = await _employeeQualificationService.SaveEmployeeQualification(employeeQualificationDto, employeeQualificationDto.EmployeeId);
+        return Ok(newQualification);
     }
 
     [Authorize(Policy = "AdminOrTalentOrJourneyOrSuperAdminPolicy")]
     [HttpPut("{id}")]
     public async Task<IActionResult> UpdateEmployeeQualification(int id, [FromBody] EmployeeQualificationDto employeeQualificationDto)
     {
-        try
-        {
-            employeeQualificationDto.Id = id;
-            var updatedQualification = await _employeeQualificationService.UpdateEmployeeQualification(employeeQualificationDto);
-            return Ok(updatedQualification);
-        }
-        catch (KeyNotFoundException knf)
-        {
-            return NotFound(knf.Message);
-        }
-        catch (Exception ex)
-        {
-            return StatusCode(500, ex.Message);
-        }
+        employeeQualificationDto.Id = id;
+        var updatedQualification = await _employeeQualificationService.UpdateEmployeeQualification(employeeQualificationDto);
+        return Ok(updatedQualification);
     }
 
     [Authorize(Policy = "AllRolesPolicy")]
```

Comment:
This endpoint seems to pull all qualifications and does not filter by anything. It seems like this endpoint is wide open and should probably have a better authorize policy?

Investigate what front-end endpoints are using this call to determine how we can manage the risk of pulling all qualifications.

```suggestion
    [Authorize(Policy = "AdminOrTalentOrJourneyOrSuperAdminPolicy")]
```

Reply:
the only instance of it on the frontend is in the actual service its not being used anywhere else currently.